argocd/utils.go

```diff
@@ -97,9 +97,17 @@ func validatePolicy(project string, role string, policy string) error {
 		return fmt.Errorf("invalid policy rule '%s': policy subject must be: '%s', not '%s'", policy, expectedSubject, subject)
 	}
 	// resource
+	// See https://github.com/muma378/argo-cd/blob/6fd4c6f44acc06f934061ce7848d72454366345b/pkg/apis/application/v1alpha1/types.go#L1555-L1561
```

Comment:
There is a comment to refactor this using a data structure `rbac.ResourceApplications` data structure maintained upstream, did you consider that instead?

Reply:
I hadn't... Glossed over it because of the related comment for `validActions` at https://github.com/argoproj/argo-cd/blob/c99669e088b5f25c8ce8faff6df25797a8beb5ba/pkg/apis/application/v1alpha1/types.go#L1527

But now that you mention it, this module won't have any "import cycle" issues. So I have gone ahead and adjusted to reference `rbacpolicy` instead of hard-coding the strings (handled actions as well while I was at it)